pkg/cri/resource-manager/requests.go

```diff
@@ -566,7 +566,7 @@ func (m *resmgr) RemoveContainer(ctx context.Context, method string, request int
 			method, container.PrettyName(), err)
 	}
 
-	m.cache.DeleteContainer(container.GetCacheID())
+	container.UpdateState(cache.ContainerStateStale)
```

Comment:
It seems to me that elsewhere the order is
1. `ReleaseResources(container)`
2. `container.UpdateState(stale)`
3. `runPostReleaseHooks(container)`

but here in `RemoveContainer()` the order (2) and (3) go the other way around.

Reply:
Please, read my comment regarding this above, in the very same message where I pinged you and marquiz...

Reply:
> but here in `RemoveContainer()` the order of (2) and (3) is the other way around.

Hmm... I'll need to recheck that. That would leave the container lingering in the cache
in a stale state and while it should have no effect on resource allocations we still don't
want that.

Reply:
@klihub , thanks, I'm completely fine with separate single-`ReleaseResources` patch to make the release sequence nice and clean in policy perspective. Sorry about reading only the code and forgetting the latter part of the message.

Reply:
@askervin Now updating the state and running the hooks should be more consistent.

Reply:
@askervin The lingering containers were a thinko introduced by the initial version of that 'don't jump the gun' commit: neither `RemoveContainer()` nor `runPostReleaseHooks()` was removing the directly passed stale removed container (released). That should be fixed now.

Reply:
> @klihub , thanks, I'm completely fine with separate single-`ReleaseResources` patch to make the release sequence nice and clean in policy perspective. Sorry about reading only the code and forgetting the latter part of the message.

No problemo. It is only good that you raised it, because it could have been as well so that I haven't had considered it. This time it just happened to be so that I have.